main/guides/zoe/proposal.md

```diff
@@ -4,69 +4,85 @@
+Keywords are unique identifiers per contract, that tie together proposals,
+payments to be escrowed, and payouts to users.
+In the example above, "Asset" and "Price" are the Keywords. However, in an auction contract,
+the Keywords might be "Asset" and "Bid".
+
+Each `AmountMath.make` call above is just making an ERTP [Amount](/reference/ertp-api/ertp-data-types.md#amount), or description of digital assets.
+In this case, `AmountMath.make(quatloosBrand, 4n)` creates a description of 4 units
+of our imaginary Quatloos currency and `AmountMath.make(moolaBrand, 15n)` creates a description
+of 15 units of our imaginary Moola currency. (The "n" appended after each number indicates that
+it is represented as a [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt)
+rather than a [Number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number))
+
+::: warning Note
+It's important to understand that Amounts are just descriptions of assets with no
+intrinsic value. [Payments](/reference/ertp-api/payment.md) hold actual digital assets.
+:::
+
+**exit** specifies how the offer can be can cancelled. It must conform to one of three shapes:
+- `{ onDemand: null }`: (Default) The offering party can cancel on demand.
+- `{ waived: null }`: The offering party can't cancel and relies entirely on the smart contract to promptly finish their offer.
+- `{ afterDeadline: deadlineDetails }`: The offer is automatically cancelled after a deadline,
+  as determined by its `timer` and `deadline` properties.
+
+For more details, see [Proposals](/reference/zoe-api/zoe.md#proposals).
 
 ## Escrowed Payments
 
-Using the same keywords as your `proposal`, you must specify a `PaymentKeywordRecord`.
-This is a record with the keywords as keys, and `payments` containing digital assets as
-values. Zoe escrows these `payments` on behalf of this offer until the offer is completed
-or rejected or the assets are reassigned to another offer. 
+Using the same Keywords as the **give** object in your **proposal**, you must specify a [PaymentKeywordRecord](/reference/zoe-api/zoe-data-types.md#keywordrecord) containing [Payments](/reference/ertp-api/payment.md) of the corresponding digital assets.
+Zoe escrows these payments on behalf of your offer until it is completed
+or rejected or the assets are reassigned to another offer.
 ```js
-const paymentKeywordRecord = { 
-  'Asset' : quatloosPayment, 
-  'Price' : moolaPayment 
+const payments = {
+  Asset: quatloosPayment,
+  Price: moolaPayment,
 };
 ```
-## Returned Value
 
-`offer()` returns a `UserSeat` object. Its name comes from the concept of "having a seat at the table" 
-for the contract's execution. 
+## Offer Arguments
+
+To pass additional arguments to the **offerHandler** contract code associated with the
+invitation, send them in an **offerArgs** record.
```

Comment:
I thought that was official JavaScript terminology.

Reply:
I believe @erights coined `record` and in his interpretation, record means const-object-as-struct. The TC-39 Records and Tuples proposal uses the word to mean new-primitive-as-immutable-struct. TypeScript’s `Record` type suggests object-as-mutable-dictionary. I believe `offerArgs` is a const-object-as-dictionary. I am distinguishing “const” from “hardened” to mean: the informal contract is that this method will not mutate it and it consequently _may_ be hardened.

It might be best to just say `Object` here.

Reply:
@gibson042 This is great. We saw places where offerArgs was typed `unknown` and others where it was typed `Object`. I think we need to also converge on the latter in the code.

Reply:
Updated.

> To pass additional arguments to the **offerHandler** contract code associated with the
> invitation, send them in an **offerArgs** [CopyRecord](/glossary/#copyrecord).